common/credsregistry/tokens.go

```diff
@@ -0,0 +1,73 @@
+package credsregistry
+
+import (
+	"time"
+
+	"golang.org/x/oauth2"
+)
+
+// GetOauthToken constructs Token group from creds file.
+// Some connectors may implement Refresh tokens, when it happens expiry must be provided alongside.
+// Library shouldn't attempt to refresh tokens if API doesn't support `refresh_token` grant type.
+func (r ProviderCredentials) GetOauthToken() *oauth2.Token {
```

Comment:
This procedure was unique to proxy script. I moved it here and every connector that needs to load AT/RT tokens will do it using this method. Should be a common thing.